src/coreclr/tools/aot/Mono.Linker.Tests/TestCasesRunner/ILCompilerDriver.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using ILCompiler;
+using ILLink.Shared.TrimAnalysis;
+using Internal.IL;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace Mono.Linker.Tests.TestCasesRunner
+{
+	public class ILCompilerDriver
+	{
+		private const string DefaultSystemModule = "System.Private.CoreLib";
+
+		public void Trim (ILCompilerOptions options, ILogWriter logWriter)
```

Comment:
@MichalStrehovsky could you please review this method? It's basically the main part of this change - this runs enough of the ILScanner to get diagnostics.

The rest of this change is mostly making stuff work in XUnit and test adaptations.

Reply:
Looks good to me.

We could potentially simplify things here:
* I think OS/arch matters at all - we could just say we're compiling for x64 Windows and be done with it - if we keep the logic to detect it, it will need fixes whenever a new OS/arch is adde)
* The list of InitAssemblies probably doesn't matter and we could skip that. JanK is deleting various System.Private assemblies so having this list just means more things to fix up.